web/packages/shared/utils/highbar.test.ts

```diff
@@ -368,3 +368,26 @@ describe('equalsDeep', () => {
     ).toBe(false);
   });
 });
+
+test('runOnce only runs once and preserves "this"', () => {
+  class Class {
+    value: symbol;
+
+    update = runOnce(
+      // Use 'this' in a convoluted way to make sure it's passed correctly.
+      function (this: Class) {
+        this.value = Symbol();
+      }
+    );
+  }
+
+  const c = new Class();
+
+  // First call updates the value (and doesn't throw because of 'this' being undefined).
+  expect(() => c.update()).not.toThrow();
+  const firstReturnValue = c.value;
+
+  // Subsequent calls return the same value.
+  c.update();
+  expect(c.value).toBe(firstReturnValue);
```

Comment:
This doesn't test the return value of `update` which `runOnce` explicitly caches, does it? 😏 We'd need to return the symbol from `update` instead of storing it in a property. But I guess you also want to test `this`, so maybe we do both? As in store the symbol on the property but also return it from `update`.

Reply:
Right, fixed!